### PR TITLE
transport tests: many streams and lots of data

### PR DESCRIPTION
**Pull request by MarcoPolo**
Related to #2194.

There is a failure with mplex that I haven't spent time yet debugging. If it's not obvious, I'd suggest we fix this in parallel with the webrtc work. Meaning merge this as is and have an issue to fix it soon.

### CODE REVIEW

**Comment by MarcoPolo** [comment]
Windows stalls with webtransport:
```
2023-05-16T19:50:01.2122591Z === RUN   TestManyBigPings/WebTransport
2023-05-16T19:50:01.2122972Z panic: test timed out after 10m0s
2023-05-16T19:50:01.2123256Z running tests:
```

**Comment by MarcoPolo** [comment]
Windows seems to perform pretty badly on QUIC and WebTransport:

**Comment by BigLep** [comment]
@MarcoPolo : I saw you closed this.  Is that because there is a replacement PR coming?

**Comment by MarcoPolo** [comment]
> @MarcoPolo : I saw you closed this. Is that because there is a replacement PR coming?

Fat fingered thanks for the flag.

**Comment by MarcoPolo** [comment]
I'd lean towards skipping windows tests for now and creating a tracking issue so as not to block WebRTC integration. Thoughts @marten-seemann ?

I'm also spinning up a Windows VM to try to investigate this a bit as well.

**Comment by MarcoPolo** [comment]
Hmm windows qlog does not look good. Large latency in the RTT measurement and the session takes ~300ms with a lot of gaps of inactivity on a recording with just 2 streams.

Here's the qlogs:
[qlogs.zip](https://github.com/libp2p/go-libp2p/files/11521463/qlogs.zip)

@marten-seemann could you take a look if anything obvious jumps out to you?

File: p2p/test/transport/transport_test.go [hunk 1]
```diff
@@ -220,12 +227,81 @@ func TestBigPing(t *testing.T) {
...
+			defer func() {
+				t.Log("Total time:", time.Since(start))
+			}()
+
+			require.NoError(t, h2.Connect(context.Background(), peer.AddrInfo{
+				ID:    h1.ID(),
+				Addrs: h1.Addrs(),
+			}))
+
+			h1.SetStreamHandler("/BIG-ping/1.0.0", func(s network.Stream) {
```
Comment (marten-seemann) [inline p2p/test/transport/transport_test.go:260]:
```suggestion
			h1.SetStreamHandler("/big-ping", func(s network.Stream) {
```

File: p2p/test/transport/transport_test.go [hunk 2]
```diff
@@ -220,12 +227,81 @@ func TestBigPing(t *testing.T) {
 	}
 }
 
+func TestManyBigPings(t *testing.T) {
```
Comment (marten-seemann) [inline p2p/test/transport/transport_test.go:230]: Can you add a comment what this is supposed to test?

File: p2p/test/transport/transport_test.go [hunk 3]
```diff
@@ -278,11 +354,111 @@ func TestManyStreams(t *testing.T) {
 	}
 }
 
+func TestManyMoreStreams(t *testing.T) {
```
Comment (marten-seemann) [inline p2p/test/transport/transport_test.go:357]: Here too. I don't really understand what's going on in this test.

File: p2p/test/transport/transport_test.go (hunk 1 above)
Comment (MarcoPolo) [reply p2p/test/transport/transport_test.go:260]: It's capital because it's BIG! I'll change though

**Pull request merged**